sky/utils/common_utils.py

```diff
@@ -473,11 +502,12 @@ def method_with_retries(*args, **kwargs):
                     time.sleep(backoff.current_backoff())
                 else:
                     raise
+        assert False, 'Unreachable'
 
     return method_with_retries
 
 
-def class_fullname(cls, skip_builtins: bool = True):
+def class_fullname(cls: Type[T], skip_builtins: bool = True) -> str:
```

Comment:
Can we just omit the type parameter?
```suggestion
def class_fullname(cls: Type, skip_builtins: bool = True) -> str:
```